websrc/gadget_menu/src/gadget_menu_main.tsx

```diff
@@ -214,6 +232,7 @@ const k_rayDir = new vec3( [ 0, 0, -1 ] );
 interface GadgetMenuSettings
 {
 	favorites: string[];
+	autoLaunch: string[];
```

Comment:
It may not be possible to know the answer to this question, but is an array of URLs the right storage here? Should this be an array of objects that have the URL and a bool for autolaunch?  If we add even one more global setting to gadgets, we'd prefer the latter. 

I'm not entirely sure what the other global settings would be. Maybe a ban bit for remote gadgets (don't show me the gadget at this URL again, even if somebody else is using it) or a ban bit for sharing a gadget of your own?

Obviously it'd be possible to convert from one to the other down the road if we decided to make a change.

Reply:
yeah I thought about that as well and went for the path of least resistance, I'm all for tackling that now as opposed to later though. It would make the indexOf things more clear as well.

Reply:
Maybe "favorite" should be a flag in that interface too?

Reply:
roger, I'll make this change, we'll need to do a guard on retrieval for updating, do you prefer using a versioned key for local storage, resetting to a default state if things don't match, or supporting upgrades from <version - 1> states.

Reply:
change made